app/main/routines/startup.ts

```diff
@@ -0,0 +1,65 @@
+const { ipcMain } = require("electron")
+
+import { config } from "app/common/config"
+import { asyncChannel, deadLetterChannel, syncChannel } from "app/common/ipc"
+import { asRuntimeType, Contexts } from "app/common/runtimeTypes"
+import { Wallets } from "app/common/runtimeTypes/storage/wallets"
+import * as api from "app/main/api"
+import * as ipc from "app/main/ipc"
+import { SubSystems } from "app/main/system"
+import * as ui from "app/main/ui"
+
+/**
+ * Startup routine.
+ * It defines the workflow that the main process follows when cold starting.
+ * @param system
+ */
+export default async function startupRoutine(system: SubSystems) {
+  const returned = await Promise.all([
+    // Initialize app state
+    stateStart(system),
+    // Initialize IPC channels
+    ipcStart(system),
+    // Start Electron UI
+    uiStart()
+  ])
+
+  return { channels: returned[1] }
+}
+
+/**
+ * Initialize app state.
+ * @param system
+ */
+async function stateStart(system: SubSystems) {
+  // Get and parse wallets from storage
+  const _wallets = system.appStorage.get("wallets")
+  const wallets = asRuntimeType(_wallets, Wallets, Contexts.STORAGE)
+  console.log(wallets)
```

Comment:
We shouldn't use `console.log`, if this is a debug log we would like to have you should use `log.debug` instead (using our `logging` module).

Also, why do we read the wallets and do nothing with them? shouldn't they be added to the app state manager?

Reply:
Oops... I left that `console.log` behind me, my bad!

As per the other issue, you are right, line `40` should read:
```ts
system.appStateManager.update({ appInfo: config.appInfo, wallets })
```